From 04978316c9295ad2c3ab7f99f5e9498cbf658772 Mon Sep 17 00:00:00 2001
From: Ihor Usyk <ihor_usyk@epam.com>
Date: Wed, 15 Feb 2023 21:20:34 +0100
Subject: [PATCH 16/18] domain Dom0: call the sci_domain_init once.

Function sci_domain_init is called from the two places:
1. construct_dom0 - for domain Dom0
2. arch_domain_create - for all domains.

Double call of sci_domain_init for dom0 assigns the two sci channels
to one domain 'Dom0'. It means that 3rd domain (DomA) can not get the sci channel
because all of them are busy.
To avoid this situation, the call of sci_domain_init in arch_domain_create is moved under
the condition !is_hardware_domain.

Signed-off-by: Ihor Usyk <ihor_usyk@epam.com>
---
 xen/arch/arm/domain.c | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/xen/arch/arm/domain.c b/xen/arch/arm/domain.c
index f0827caeae..1dbc485623 100644
--- a/xen/arch/arm/domain.c
+++ b/xen/arch/arm/domain.c
@@ -765,14 +765,12 @@ int arch_domain_create(struct domain *d,
         /* At this stage vgic_reserve_virq should never fail */
         if ( !vgic_reserve_virq(d, GUEST_EVTCHN_PPI) )
             BUG();
-    }
-
-    if ( !is_idle_domain(d) && 
-            config->arch.arm_sci_type != XEN_DOMCTL_CONFIG_ARM_SCI_NONE )
-    {
-	if ( (rc = sci_domain_init(d, config->arch.arm_sci_type,
-		&config->arch)) != 0)
-		goto fail;
+        if ( config->arch.arm_sci_type != XEN_DOMCTL_CONFIG_ARM_SCI_NONE )
+        {
+            if ( (rc = sci_domain_init(d, config->arch.arm_sci_type,
+                                        &config->arch)) != 0)
+                goto fail;
+        }
     }
 
     /*
-- 
2.25.1

